packages/providers/src/transaction-response/transaction-response.ts

```diff
@@ -203,7 +200,9 @@ export class TransactionResponse {
     for await (const { statusChange } of this.provider.operations.statusChange({
       transactionId: this.id,
     })) {
-      if (statusChange.__typename !== 'SubmittedStatus') break;
+      if (statusChange.type !== 'SubmittedStatus') {
+        break;
+      }
```

Comment:
@nedsalk why not dispose of the connection manually?

 ```ts
       if (statusChange.type !== 'SubmittedStatus') {
         this.provider.subscriptionClient.dispose();
         break;
      }
 ```
 
 I've tried this locally and this seems to solve the problem while all tests continue to pass without having to set a value for `openHandlesTimeout` on jest config.

Reply:
That is disposing of the whole client, not just one connection. The client might have multiple subscriptions running at one time and disposing it would end all of them, not just the one that we want.

Reply:
A question:

If that is the case, why calling `this.provider.subscriptionClient.dispose();` inside this `if` statement does not make any of the tests fail?

Another thing, the loop's if statement was wrong and therefore was always evaluated as true. This implies that the code doesn’t wait for the status property to update as expected. 

However, despite this apparent logical error, all tests still pass. Could there be missing or ineffective tests that fail to catch this issue? Are there other reasons why the tests are not failing as expected with the wrong logic in place?

Reply:
> If that is the case, why calling this.provider.subscriptionClient.dispose(); inside this if statement does not make any of the tests fail?

Because there are no other active subscriptions on that specific subscription client which would be affected by the disposal.

> Another thing, the loop's if statement was wrong and therefore was always evaluated as true. This implies that the code doesn’t wait for the status property to update as expected.

> However, despite this apparent logical error, all tests still pass. Could there be missing or ineffective tests that fail to catch this issue? Are there other reasons why the tests are not failing as expected with the wrong logic in place?

All of the tests were succeeding because the node was so fast in processing the transaction and changing its status from `SubmittedStatus` to `SuccessStatus` that the call to the `statusChange` subscription didn't make any difference.

A proper test that would've 100% failed and caught this bug is the one that you created [here](https://github.com/FuelLabs/fuels-ts/pull/1302#discussion_r1341645141) when you found the issue. You forced the node to slow down, which brought out the wrong logic to the forefront. This test is missing and I will write it by using `launchNode` with settings similar to the ones you used when you found the bug.

Reply:
@Torres-ssf I implemented a test for it (found and existing one and reimplemented with `launchNode`).